apps/mobile/src/libs/components/typing-text-view/typing-text-view.tsx

```diff
@@ -0,0 +1,77 @@
+import { Text, View } from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import {
+	fontWeightToFamily,
+	presetToTextStyle,
+	sizeToTextStyle,
+} from "~/libs/maps/maps";
+import {
+	type StyleProp,
+	type TextStyle,
+	type ValueOf,
+} from "~/libs/types/types";
+
+import {
+	DEFAULT_START_TYPING_DELAY,
+	FIRST_LETTER_INDEX,
+	LETTERS_PRINTED_PER_STEP,
+	NEXT_LETTERS_PRINTED_DELAY,
+} from "./libs/constants/constants";
+import { useTypingText } from "./libs/hooks/hooks";
+
+type Properties = {
+	children: string;
+	lettersPrintedPerStep?: number;
+	maskingColor?: ValueOf<typeof BaseColor>;
+	nextLettersPrintedDelay?: number;
+	preset?: keyof typeof presetToTextStyle;
+	size?: keyof typeof sizeToTextStyle;
+	startTypingDelay?: number;
+	style?: StyleProp<TextStyle>;
+	textColor: ValueOf<typeof BaseColor>;
+	weight?: keyof typeof fontWeightToFamily;
+};
+
+const TypingTextView: React.FC<Properties> = ({
+	children,
+	lettersPrintedPerStep = LETTERS_PRINTED_PER_STEP,
+	maskingColor = BaseColor.BG_WHITE,
+	nextLettersPrintedDelay = NEXT_LETTERS_PRINTED_DELAY,
+	preset = "default",
+	size,
+	startTypingDelay = DEFAULT_START_TYPING_DELAY,
+	style,
+	textColor = BaseColor.BLACK,
+	weight,
+}: Properties) => {
+	const textStyles: StyleProp<TextStyle> = [
+		presetToTextStyle[preset],
+		size && sizeToTextStyle[size],
+		weight && fontWeightToFamily[weight],
+		style,
+	];
+
+	const hiddenTextColorStyle = { color: maskingColor };
+	const visibleTextColorStyle = { color: textColor };
+
+	const { hiddenLetters, visibleLetters } = useTypingText({
+		firstLetterIndex: FIRST_LETTER_INDEX,
+		lettersPrintedPerStep,
+		nextLettersPrintedDelay,
+		startTypingDelay,
+		text: children,
+	});
+
+	return (
+		<View>
```

Comment:
Also if don't set an styles to that wrapper view, we can use Fragment instead
```suggestion
		<>
```